core/asset.go

```diff
@@ -94,3 +95,9 @@ func (a *Asset) Deserialize(r io.Reader) error {
 	}
 	return nil
 }
+
+func (a *Asset) Hash() common.Uint256 {
+	buf := new(bytes.Buffer)
+	a.Serialize(buf)
+	return common.Uint256(common.Sha256D(buf.Bytes()))
```

Comment:
Try just return common.Sha256D(buf.bytes())

Reply:
modified and refreshed the commit